Client/src/Components/Charts/Gauge/index.jsx

```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import { RadialBarChart, RadialBar, ResponsiveContainer } from "recharts";
+import PropTypes from "prop-types";
+
+const DATA = [{ value: 100 }];
+const PROGRESS_THRESHOLD = 50;
+const DEFAULT_CONTAINER_HEIGHT = 160;
+const DEFAULT_HEADER_FONT = { fontSize: "11px" };
+const DEFAULT_SUBHEADER_FONT = { fontSize: "9px" };
+const TEXT_POSITIONS = {
+	value: { x: "50%", y: "45%" },
+	label: { x: "50%", y: "55%" },
+};
+const COMMON_RADIALBAR_PROPS = {
+	minAngle: 15,
+	clockWise: true,
+	dataKey: "value",
+	cornerRadius: 0,
+};
+
+const RADIALBARCHART_PROPS = {
+	cx: "50%",
+	cy: "50%",
+	innerRadius: "45%",
+	outerRadius: "100%",
+	startAngle: 90,
+	endAngle: -270,
+};
+
+const RADIALBAR_OVERLAY_PROPS = {
+	transform: "rotate(180deg)",
+	position: "absolute",
+	top: "0",
+	left: "0",
+};
+
+const Gauge = ({
+	progressValue,
+	displayText,
+	theme,
+	containerHeight,
+	gaugeHeader,
+	gaugeSubheader,
+}) => {
+	const header = gaugeHeader ?? DEFAULT_HEADER_FONT;
+	const subheader = gaugeSubheader ?? DEFAULT_SUBHEADER_FONT;
+	const myProgressValue =
+		progressValue < 0 ? 0 : progressValue > DATA[0].value ? DATA[0].value : progressValue;
```

Comment:
Is 0 minimum value, and 100 maximum value? If so, it would be nice to store them as constants. 
And to avoid the nested ternaries, we could go with:

```
const validProgressValue = Math.max(MINIMUM_VALUE, Math.min(progressValue, MAXIMUM VALUE));
```

Reply:
I see, actually that is a better way to check min and maz, will change, thanks!